proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -313,7 +315,7 @@ public ProctorResult determineTestGroups(
                 final String invalidIdentifierMessage =
                         String.format(
                                 "An invalid identifier '%s' for test type '%s'"
-                                        + " was detected. Using fallback buckets for the test type.",
+                                        + " was detected. Using fallback buckets for the test type unless unitless allocation is enabled.",
```

Comment:
Which reminds me we may also want to take care about the warning logging added to the validate() method in proctor-internal (https://link.indeed.tech/GLY39NXAE)

Reply:
To add that check might be a more significant impact on performance. 

This for loop is:

For each test type present in the application:
 ->  check if the identifier in a map is valid otherwise log

It currently does not access test definitions. Also to correctly consider all cases the code would have to check every test per test type to make sure every test has unitless allocations enabled when an invalid identifier is present.

Reply:
Ok, that makes sense. It still feels potentially annoying to application owners if the library logs all those warnings but in a way it's not a blocker.
An idea to explore would be to log it to logrepo instead of the application logs. That way we can still detect applications which have potential issues with their identifiers.  (in a separate ticket that can be done at a different time).

Reply:
Just an idea, if we want to limit the log volume, we may do something similar to what Zack's has done before, deduping the logs.
https://github.com/indeedeng/proctor/pull/93/files